api/v1alpha1/sink_types.go

```diff
@@ -69,7 +69,8 @@ type SinkSpec struct {
 	Pod PodPolicy `json:"pod,omitempty"`
 
 	// +kubebuilder:validation:Required
-	Messaging `json:",inline"`
+	Messaging    `json:",inline"`
+	TLSTrustCert CryptoSecret `json:"tlsTrustCert,omitempty"`
```

Comment:
Lets not reuse the `CryptoSecret` struct here, because it is for E2E encryption, and we may add additional feature `asEnv` later, which may not prefect for TLS cert mounting. And I would like to suggest adding a `TLSTrustCert` section in `PulsarMessaging` to reference the TLS cert secret and use it when TLSSecret does have `tls_trust_cert_path` specified.